### ISSUE TITLE
Bump to 2.2.19 BREAKS usage without "options.authKey'

### ISSUE THREAD

**Issue by veikovx** [neumino/rethinkdbdash#238]
Hi

With the new release  2.2.19, running without setting `options.authKey` is broken.

Errors same as shown here: https://app.wercker.com/#buildstep/570f1c87f81dabb82705d323

authKey is set in index` r.prototype._authKey = '';` and this passes the new `if (options.authKey != null)` test.

**Comment by soplakanets** [comment]
+1

Here is my stacktrace:

```
TypeError: must start with number, buffer, array or string
    at fromObject (buffer.js:141:11)
    at new Buffer (buffer.js:62:10)
    at Socket.<anonymous> (/opt/roveme/search/node_modules/roveme-lib-rethinkdb/node_modules/thinky/node_modules/rethinkdbdash/lib/connection.js:113:22)
    at emitNone (events.js:72:20)
    at Socket.emit (events.js:166:7)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1044:10)
```

**Comment by cggaurav** [comment]
+1

**Comment by cggaurav** [comment]
Have a default for self.authkey if we are using it or not define it at all? Broke our environments today

**Comment by samuelsoderberg** [comment]
I got a similar issue running Debian:
```
    Creating a pool connected to 127.0.0.1:28015
    Setting up listener...
    buffer.js:247
      throw new TypeError(kFromErrorMsg);
      ^
    
    TypeError: must start with number, buffer, array or string
        at fromObject (buffer.js:247:9)
        at Function.Buffer.from (buffer.js:115:10)
        at new Buffer (buffer.js:94:17)
        at Socket.<anonymous> (/home/samuel/Code/app/node_modules/rethinkdbdash/lib/connection.js:113:22)
        at emitNone (events.js:85:20)
        at Socket.emit (events.js:179:7)
        at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1046:10)
```

After trying different rethinkdb-versions (both the standalone server and the package.json-dependency), trying different versions of both npm and node etc, I installed Ubuntu in pure frustration... But the problem persists.

My code:

```
    function connect() {
      return r.connect(config.get('rethinkdb'));
    }
    
    export function liveUpdates(io) {
      console.log('Setting up listener...');
      connect()
      .then(conn => {
        r
        .table('tickets')
        .changes().run(conn, (err, cursor) => {
          console.log('Listening for changes...');
          cursor.each((err, change) => {
            console.log('Change detected', change);
            io.emit('ticket-change', change);
          });
        });
      });
    }

```
npm -v

3.8.3

node -v

v5.10.1

rethinkdb --version

rethinkdb 2.3.0~0wily (GCC 5.2.1)

**Comment by mbroadst** [comment]
bananas this was released, what's the point of having CI if its not used to verify the code before releasing (in particular a patch release)

**Comment by rsantosdev** [comment]
It works well on all my project tests when using rethinkdb:latest, but fails all tests when using rethinkdb:2.2

**Comment by mbroadst** [comment]
This has nothing to do with rethinkdb version (though the changes themselves are likely to be part of the refactoring to support 2.3.x's new auth handshake, perhaps mistakenly committed), rather that a Buffer is being created with (probably) `undefined` here: https://github.com/neumino/rethinkdbdash/blob/master/lib/connection.js#L113

**Comment by rsantosdev** [comment]
It works well on all my project tests when using rethinkdb:latest, but fails all tests when using rethinkdb:2.2

**Comment by neumino** [comment]
My bad, I accidentally introduce a commit for RethinkDB 2.3 in the last
commit. I'll fix that tonight, please us 2.2.18 for now.

**Comment by neumino** [comment]
Just released 2.2.20 with the fix.
